components/locid/src/locale.rs

```diff
@@ -149,6 +149,55 @@ impl Locale {
         self.extensions.unicode.keywords.get(key)
     }
 
+    /// Mutating setter method for Unicode extensions.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu::locid::{extensions::unicode::Key, locale, unicode_ext_key};
+    ///
+    /// const KEY: Key = unicode_ext_key!("hc");
+    ///
+    /// let mut loc = locale!("en-US");
+    /// loc.set_unicode_extension(KEY, "h12".parse().expect("valid subtag"));
+    /// assert_eq!(
+    ///     loc,
+    ///     "en-US-u-hc-h12"
+    /// );
+    /// ```
+    pub fn set_unicode_extension(
```

Comment:
Note: We already had `Locale::get_unicode_extension` hoisted to the top level, so I thought it would be acceptable to hoist the setters up, too.

Reply:
Ugh, I think it slipped for me. Can we not? Could we revert that and have `locale.extensions.unicode.set/get`?

Reply:
Forensics: the existing `Locale::get_unicode_extension`, which I used as a model, was added here:

https://github.com/unicode-org/icu4x/pull/803